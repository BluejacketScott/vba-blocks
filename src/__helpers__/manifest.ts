import { dir as FIXTURES } from '../../tests/__fixtures__';
import { Manifest } from '../manifest';
import { isPathDependency } from '../manifest/dependency';
import { relative } from '../utils/path';

export function normalizeManifest(manifest: Manifest, relativeTo: string = FIXTURES): Manifest {
  for (const src of manifest.src) {
    src.path = normalizePath(src.path, relativeTo);

    if (src.binary) {
      src.binary = normalizePath(src.binary, relativeTo);
    }
  }
  for (const dependency of manifest.dependencies) {
    if (isPathDependency(dependency)) {
      dependency.path = normalizePath(dependency.path, relativeTo);
    }
  }
  for (const src of manifest.devSrc) {
    src.path = normalizePath(src.path, relativeTo);

    if (src.binary) {
      src.binary = normalizePath(src.binary, relativeTo);
    }
  }
  for (const dependency of manifest.devDependencies) {
    if (isPathDependency(dependency)) {
      dependency.path = normalizePath(dependency.path, relativeTo);
    }
  }

  if (manifest.target) {
    manifest.target.path = normalizePath(manifest.target.path, relativeTo);
  }

  return manifest;
}

function normalizePath(path: string, relativeTo: string = FIXTURES): string {
  return relative(relativeTo, path);
}
