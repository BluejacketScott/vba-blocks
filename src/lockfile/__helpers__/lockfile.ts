export function normalizeLockfile(lockfile: string): string {
	return lockfile.replace(/vba\-blocks v\d.\d.\d/g, "vba-blocks v#.#.#");
}
