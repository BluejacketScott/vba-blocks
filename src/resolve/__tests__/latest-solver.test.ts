import solve from '../latest-solver';
import Resolver from '../resolver';
import { getConfig } from '../../../tests/__helpers__';
import * as workspaces from '../../../tests/__fixtures__/workspaces';

test('solves simple tree', async () => {
  const config = getConfig();
  const resolver = new Resolver(config);

  const solution = await solve(workspaces.simple, resolver);

  expect(solution).toMatchSnapshot();
});

test('solves complex tree', async () => {
  const config = getConfig();
  const resolver = new Resolver(config);

  const solution = await solve(workspaces.complex, resolver);

  expect(solution).toMatchSnapshot();
});

test('fails to solve needs-sat tree', async () => {
  const config = getConfig();
  const resolver = new Resolver(config);

  await expect(solve(workspaces.needsSat, resolver)).rejects.toMatchSnapshot();
});

test('fails to solve unresolvable tree', async () => {
  const config = getConfig();
  const resolver = new Resolver(config);

  await expect(
    solve(workspaces.unresolvable, resolver)
  ).rejects.toMatchSnapshot();
});
