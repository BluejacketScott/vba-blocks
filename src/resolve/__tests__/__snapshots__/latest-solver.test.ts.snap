// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fails to solve needs-sat tree 1`] = `[Error: Unable to resolve dependency graph for given manifest]`;

exports[`fails to solve unresolvable tree 1`] = `[Error: Unable to resolve dependency graph for given manifest]`;

exports[`solves complex tree 1`] = `
Array [
  Object {
    "dependencies": Array [
      Object {
        "name": "json",
        "registry": "vba-blocks",
        "version": "^2.3.0",
      },
    ],
    "id": "web@4.1.5",
    "name": "web",
    "source": "registry+vba-blocks#sha256-56c0bfaa84be448b39c65aa83f1c7530db0635a6b2b6ee8589a18feea29e37e9",
    "version": "4.1.5",
  },
  Object {
    "dependencies": Array [
      Object {
        "name": "dictionary",
        "registry": "vba-blocks",
        "version": "^1",
      },
    ],
    "id": "json@2.3.0",
    "name": "json",
    "source": "registry+vba-blocks#sha256-3c2760ce2940dd19c16542cead792e0d05e58566155b1fbc72d445fe50a36799",
    "version": "2.3.0",
  },
  Object {
    "dependencies": Array [],
    "id": "dictionary@1.4.1",
    "name": "dictionary",
    "source": "registry+vba-blocks#sha256-b241062bcd7ffd68c9063e3980a5df4c6781df29d96763a020d291e52454f6a5",
    "version": "1.4.1",
  },
]
`;

exports[`solves standard tree 1`] = `
Array [
  Object {
    "dependencies": Array [
      Object {
        "name": "dictionary",
        "registry": "vba-blocks",
        "version": "^1",
      },
    ],
    "id": "json@2.3.0",
    "name": "json",
    "source": "registry+vba-blocks#sha256-3c2760ce2940dd19c16542cead792e0d05e58566155b1fbc72d445fe50a36799",
    "version": "2.3.0",
  },
  Object {
    "dependencies": Array [],
    "id": "dictionary@1.4.1",
    "name": "dictionary",
    "source": "registry+vba-blocks#sha256-b241062bcd7ffd68c9063e3980a5df4c6781df29d96763a020d291e52454f6a5",
    "version": "1.4.1",
  },
]
`;
